starter-code/lib/harryPotter.js

```diff
@@ -8,3 +8,26 @@
                    "4 October", "Filius Flitwick", "17 October", "Molly Weasley",
                    "30 October", "Bill Weasley", "29 November", "Rubeus Hagrid",
                    "6 December", "Charlie Weasley", "12 December"];
+
+
+outputArray = [[]];
+
+
+function sepate(array){ 
+    outputArray = array.filter(function(element){
+
+
+    });
+}
+
+
+
+var birthdayPairs = birthdays.reduce( function (accumulator, item, index) {
```

Comment:
Very good!

I would recommend you use the fourth argument of the reduce method, array. Here is the complete solution.

```
const birthDayNamePairsArray = allBirthdays.reduce(function (accumulator, currentValue, ix, array) {
    if (ix % 2 === 0) {
      accumulator.push([array[ix], array[ix + 1]]);
    }
    return accumulator;
  },[]);
console.log(birthDayNamePairsArray);
```

Reply:
You can also use forEach
```
// Using forEach
var nameBirtPairs = [];
birthdays.forEach(function(element, index, array) {
  if (index % 2 === 0) {
    nameBirtPairs.push([array[index], array[index + 1]]);
  }
});
console.log(nameBirtPairs);

// As we do not want any unnecessary variable in the global scope! We should wrap the forEach method inside a function, like this:
function generateNameBirth(array) {
  var nameBirtPairs = [];
  array.forEach(function(element, index, array) {
    if (index % 2 === 0) {
      nameBirtPairs.push([array[index], array[index + 1]]);
    }
  });
  return nameBirtPairs;
}
console.log(generateNameBirth(birthdays));
```